backend/ops_api/ops/utils/authorization.py

```diff
@@ -11,17 +15,18 @@ class BasicAuthorizationPrivider(AuthorizationProvider):
     def __init__(self, authirized_users: list[str] = []):
         self.authorized_users = authirized_users
 
-    def is_authorized(self, oidc_id: str, permission: list[str]) -> bool:
-        # Perform some basic user lookup and permission check
-        # user = User.query.filter_by(oidc_id=id).first()
-        # if user is not None:
-        #     for role in user.roles:
-        #         if permission in role.permissions.split(",");
-        #             return True
-        # return False
-
-        # TODO: While we flush this out, EVERYONE GETS ACCESS!!
-        return True
+    def is_authorized(self, oidc_id: str, permissions: list[str]) -> bool:
+        stmt = select(User).where(User.oidc_id == oidc_id)
+        users = current_app.db_session.execute(stmt).all()
+        if users and len(users) == 1:
+            user = users[0][0]
+
+            for role in user.roles:
+                for permission in permissions:
+                    if permission in role.permissions:
+                        return True
```

Comment:
Do we want to ensure the User has all `permissions` here rather than one of them?

Reply:
Well, I was seeing permissions a little differently - so if an endpoint has 2 permissions, like GET_USER and GET_USERS, then that means the endpoint allows for both options, not restricting someone to needing both to use that endpoint.

The way you're describing, it would be more and more restrictive on the endpoint, rather than permissive. That might be worth talking about as a team to come up with a consensus.

Reply:
I decided for now to make it 1 permission per endpoint. As we didn't need anything more than that. If we ever do, we can revisit this question and come with a good solution.